catroid/src/androidTestPhiro/java/org/catrobat/catroid/uiespresso/ui/activity/PhiroMyProjectsActivityTest.java

```diff
@@ -0,0 +1,148 @@
+ */
+
+package org.catrobat.catroid.uiespresso.ui.activity;
+
+import android.support.test.InstrumentationRegistry;
+import android.support.test.espresso.ViewInteraction;
+import android.support.test.runner.AndroidJUnit4;
+import android.util.Log;
+
+import org.catrobat.catroid.ProjectManager;
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.content.Project;
+import org.catrobat.catroid.content.Script;
+import org.catrobat.catroid.content.bricks.Brick;
+import org.catrobat.catroid.content.bricks.NoteBrick;
+import org.catrobat.catroid.exceptions.CompatibilityProjectException;
+import org.catrobat.catroid.exceptions.LoadingProjectException;
+import org.catrobat.catroid.exceptions.OutdatedVersionProjectException;
+import org.catrobat.catroid.io.StorageHandler;
+import org.catrobat.catroid.phiro.ui.PhiroMainMenuActivity;
+import org.catrobat.catroid.uiespresso.util.BaseActivityInstrumentationRule;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static junit.framework.Assert.assertTrue;
+
+import static org.catrobat.catroid.uiespresso.util.matchers.AdapterViewMatchers.withAdaptedDataAt;
+
+@RunWith(AndroidJUnit4.class)
+public class PhiroMyProjectsActivityTest {
+
+	private static final String TAG = PhiroMyProjectsActivityTest.class.getSimpleName();
+	@Rule
+	public BaseActivityInstrumentationRule<PhiroMainMenuActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(PhiroMainMenuActivity.class, true, false);
+
+	@Before
+	public void setUp() throws Exception {
+		removePhiroPrograms();
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	@Test
+	public void phiroProgramsInitiallyLoadedTest() {
+		onView(withText(R.string.main_menu_programs))
+				.perform(click());
+
+		ViewInteraction viewInteraction = onView(withId(android.R.id.list));
+
+		Integer position = 0;
+		for (String program : PhiroMainMenuActivity.phiroProjects) {
+			viewInteraction
+					.check(matches(withAdaptedDataAt(program, ++position)));
+		}
+	}
+
+	@Test
+	public void phiroProgramsRestoredWhenDeletedTest() {
+		onView(withText(R.string.main_menu_programs))
+				.perform(click());
+
+		StorageHandler.getInstance().deleteProject(PhiroMainMenuActivity.phiroProjects[0]);
+
+		onView(withId(android.R.id.home))
+				.perform(click());
+
+		onView(withText(R.string.main_menu_programs))
+				.perform(click());
+
+		onView(withId(android.R.id.list))
+				.check(matches(withAdaptedDataAt(PhiroMainMenuActivity.phiroProjects[0], 0)));
+	}
+
+	@Test
+	public void phiroProgramsUnchangedWhenModifiedTest() {
```

Comment:
write a separate test class that sets up a project, and then checks if it is properly displayed, and add/remove bricks there.